### ISSUE TITLE
PIL.JpegImagePlugin in _getexif TypeError: 'unicode' object cannot be interpreted as an index

### ISSUE THREAD

**Issue by sergei-maertens** [python-pillow/Pillow#1359]
This seems to be related to #635, #1163, #1256, #520.

Pillow 2.9.0, Python 2.7

Traceback:
```
Stacktrace (most recent call last):

  File "django/core/handlers/base.py", line 164, in get_response
    response = response.render()
  File "django/template/response.py", line 158, in render
    self.content = self.rendered_content
  File "django/template/response.py", line 135, in rendered_content
    content = template.render(context, self._request)
  File "django/template/backends/django.py", line 74, in render
    return self.template.render(context)
  File "django/template/base.py", line 209, in render
    return self._render(context)
  File "django/template/base.py", line 201, in _render
    return self.nodelist.render(context)
  File "django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "django/template/base.py", line 917, in render_node
    return node.render(context)
  File "django/template/loader_tags.py", line 135, in render
    return compiled_parent._render(context)
  File "django/template/base.py", line 201, in _render
    return self.nodelist.render(context)
  File "django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "django/template/base.py", line 917, in render_node
    return node.render(context)
  File "django/template/base.py", line 903, in render
    bit = self.render_node(node, context)
  File "django/template/base.py", line 917, in render_node
    return node.render(context)
  File "sorl/thumbnail/templatetags/thumbnail.py", line 59, in render
    return self._render(context)
  File "sorl/thumbnail/templatetags/thumbnail.py", line 137, in _render
    thumbnail = get_thumbnail(file_, geometry, **options)
  File "sorl/thumbnail/shortcuts.py", line 8, in get_thumbnail
    return default.backend.get_thumbnail(file_, geometry_string, **options)
  File "sorl/thumbnail/base.py", line 125, in get_thumbnail
    thumbnail)
  File "sorl/thumbnail/base.py", line 157, in _create_thumbnail
    image = default.engine.create(source_image, geometry, options)
  File "sorl/thumbnail/engines/base.py", line 20, in create
    image = self.orientation(image, geometry, options)
  File "sorl/thumbnail/engines/base.py", line 45, in orientation
    return self._orientation(image)
  File "sorl/thumbnail/engines/pil_engine.py", line 71, in _orientation
    exif = image._getexif()
  File "PIL/JpegImagePlugin.py", line 390, in _getexif
    return _getexif(self)
  File "PIL/JpegImagePlugin.py", line 424, in _getexif
    file.seek(exif[0x8769])
```
The exif data seems to be of the format
```
'Exif\x00\x00II*\x00\x08\x00\x00\x00\x07\x00\x1a\x01\x05\x00\x01\x00\x00\x00b\x00\x00\x00\x1b\x01\x05\x00\x01\x00\x00\x00j\x00\x00\x00(\x01\x03\x00\x01\x00\x00\x00\x02\x00\x00\x001\x01\x02\x00\x1d\x00\x00\x00r\x00\x00\x002\x01\x02\x00\x14\x00\x00\x00\x90\x00\x00\x00\x13\x02\x03\x00\x01\x00\x00\x00\x01\x00\x00\x00i\x87\x02\x00\x9a\x00\x00\x00\xa4\x00\x00\x00>\x01\x00\x00\xd0\x07\x00\x00\x04\x00\x00\x00\xd0\x07\x00\x00\x04\x00\x00\x00Adobe Photoshop Elements 5.0\x00\x002010:08:15 13:58:09\x00\t\x00\x00\x90
```
which is obviously encoded.

The image in question is https://modelbrouwers.nl/media/albums/44/3370/DSC_4458.jpg

**Comment by radarhere** [comment]
Perhaps there is a way to interpret the data usefully, but I suspect the most general solution to this is to catch the TypeError. I've created a PR accordingly.

**Comment by radarhere** [comment]
Is the image one that could potentially be included in the test suite under our license?

**Comment by sergei-maertens** [comment]
I've mentioned it on the cropping library, only later I realized this wasn't the most appropriate place to put the issue.

You can use the image.

**Comment by radarhere** [comment]
Thanks. I've added a test to the PR with the image.

**Comment by radarhere** [comment]
The TypeError has now been caught and merged in with PR #1414
